src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Serialization/RazorProjectInfoFileSerializer.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using MessagePack;
+using MessagePack.Resolvers;
+using Microsoft.AspNetCore.Razor.ProjectSystem;
+using Microsoft.AspNetCore.Razor.Serialization.MessagePack.Resolvers;
+
+namespace Microsoft.CodeAnalysis.Razor.Serialization;
+
+internal class RazorProjectInfoFileSerializer : IRazorProjectInfoFileSerializer
```

Comment:
Nit: When I've been doing this, I've been doing abstract classes in Workspaces, and an impl in the language server. Granted, its an empty impl, but it felt like if we have a rule of no MEF exports from workspaces, maybe we should also not put anything from workspaces in the service collection either?

Reply:
I get where you're coming from, but I think it's unnecessarily cumbersome. We don't export MEF from Workspaces because MEF registration is declarative and errant services can be pulled into MEF containers unintentionally. However, that isn't the case for the language server. If there's a fully concrete type that doesn't require any language server-specific implementation, I think it's fine to just add that concrete type to the service collection rather than creating a gratuitous abstract class. Obviously, there'll need to be a tiny implementation in VS because of MEF, but version in the language server would serve no real purpose and be completely superfluous.

Note: `ProjectSnapshotManager` is very similar. It does not need a purely abstract class just to satisfy this requirement in the language server. Today, there's an implementation in VS that's needed for MEF registration, and the fully concrete `ProjectSnapshotManager` from Workspaces is simply added to the service collection in the language server.